### DISCUSSION TITLE
How to pass a text file to fabric

### DISCUSSION THREAD

**Discussion by poezie** [danielmiessler/Fabric#918] [Q&A]
So I have used pbpaste. yt and echo to pass information to fabric , does anyone know how to pass a file ? for example I have a document or log file I would like to analyze is there a way to pass a pdf, docx, xls, or txt to fabric to run patterns agains that information ?

**Comment by SlicksRoboto** [comment]
I am also curious about this.

How do I pass a log file to fabric? Cant be so that I need to copy paste all info from the logs.


PLease advise.

**Reply by pedroibl** [reply]
1. Copy the logs you wish to analyze.
2. Paste them into a Markdown file and save it in the `contexts` directory (e.g., `logs.md`).
path: .config/fabric/contexts

4. After saving the file, execute the following command in your terminal:

fabric -C logs.md | analyze_logs -c

**Comment by SlicksRoboto** [comment]
I don´t know if this helps but I did this for analyzing logs in windows/powershell:

Get-Content .\path\to\logfile.log* | --pattern analyze_logs and it worked (*obviously you have to refer to your own path structure)
So you need to reference to the folder/files you want to have analyzed based on what system you are using and you are good to go.

I do agree that there should be more examples on how to use the different patterns, in all systems (mac, win, lin)

**Comment by snafu4** [comment]
Maybe I'm missing something but it appears you are making this far more complicated that it needs to be.
```
cat <filename>
```
will stream/show contents of the file `<filename>` in powershell or linux
```
fabric -p summarize
```
will summarize anything that is stream/sent to it so:
```
cat <filename> | fabric -p summarize
```
will summarize `<filename>`

**Reply by arcy24** [reply]
-p analyze_logs does a great job as well!

**Reply by gchingon** [reply]
you can take it a step further and have fabric dump into a markdown file, to keep/review later

`cat <filename> | fabric -p summarize > path/to/logfile-summary.md`

**Comment by bledburn** [comment]
> So I have used pbpaste. yt and echo to pass information to fabric , does anyone know how to pass a file ? for example I have a document or log file I would like to analyze is there a way to pass a pdf, docx, xls, or txt to fabric to run patterns agains that information ?

I believe snafu4 accurately detected the root cause. However, there is one deeper issue you need to understand. Pdf's Docx's, Xls's, all these file formats are not plain text. LLMs, not just through using the Fabric CLI tool, accept plain text. They need to be pre- processed into plain text (there are countless methods out there now). When a new service pops up that allows for their LLMs or AI's to interact with your Pdf's and Doc's, what they are doing under the hood is extracting the information from those formats into plain text before feeding the payloads to an LLM.
